wolfcrypt/src/wc_port.c

```diff
@@ -3344,3 +3344,176 @@ char* mystrnstr(const char* s1, const char* s2, unsigned int n)
         #include <wolfcrypt/src/port/arm/cryptoCellHash.c> /* sha256 */
     #endif
 #endif
+
+
+#ifndef SINGLE_THREADED
+
+#ifdef _MSC_VER
+    int wolfSSL_NewThread(THREAD_TYPE* thread,
```

Comment:
Wow, thread helpers! 
Note: We have some API's for this already in async here. Feel free to pull over and leverage any of it: https://github.com/wolfSSL/wolfAsyncCrypt/blob/master/wolfcrypt/src/async.c#L865

Reply:
Can we make it VERY clear this is only required for CRL monitor? We don't normally need to port threading, but this will be very handy. I'd also like to see this for `WOLFSSL_CMSIS_RTOSv2` eventually.

Reply:
Ideally I would like to use this threading interface in our unit tests. This will make them easier to port to new platforms and allow us to test TLS on more platforms.

I kept this threading interface extremely simple to make it easier to port. The async code make much more advanced usage of threading. I think it makes sense to keep them separate for now. Maybe in the future it would make sense to have an additional `THREAD_OPTIONS` parameter to allow for setting all of the advanced options being consumed by async. For now I want to keep it simple and port the tests to this common interface.